app/Topology.hs

```diff
@@ -21,23 +23,46 @@ data TopologyInfo = TopologyInfo {
   , topologyFile :: FilePath
   }
 
--- | IPv4 address with port number
+-- | IPv4 or IPv6 address with port number
 --
--- TODO: this type should be extended to take into account IPv6 addresses.
---
-data NodeAddress = NodeAddress HostName ServiceName
+data NodeAddress = NodeAddress {
+      naHostAddress :: !IP.IP
+    , naPort        :: !PortNumber
+    }
```

Comment:
This is all good stuff in this module. I wonder if some of it makes more sense to live in the network library. No specific suggestions, but keep it in mind.

Reply:
Right, at the moment this is the only application that is using this, but this may change in the future.